auth-tokens/src/main/java/com/palantir/tokens/auth/UnverifiedJsonWebToken.java

```diff
@@ -110,13 +108,13 @@ private static int countCharacter(String input, char toCount) {
      * An anticipated use of this class is making a best-effort user id extraction for logging.
      */
     public static UnverifiedJsonWebToken of(BearerToken token) {
-        List<String> segments = Splitter.on('.').splitToList(token.getToken());
+        String[] segments = token.getToken().split("\\.");
         Preconditions.checkArgument(
-                segments.size() == 3,
+                segments.length == 3,
                 "Invalid JWT: expected 3 segments",
-                SafeArg.of("segmentsCount", segments.size()));
+                SafeArg.of("segmentsCount", segments.length));
```

Comment:
this is a hot path, worth inverting this into `if (segments !=3) { /*work + throw */}`?

Reply:
won't the JVM figure this out eventually?

Reply:
Not likely, especially if we ever encounter a bad input.

Reply:
alright, changed